modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -0,0 +1,218 @@
+
+/**
+ * The replica service.
+ */
+public class ReplicaManager implements IgniteComponent {
+    /** Busy lock to stop synchronously. */
+    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    /** Prevents double stopping the component. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
+    /** Cluster network service. */
+    private final ClusterService clusterNetSvc;
+
+    /** Partition storage. */
+    private final MvPartitionStorage mvDataStorage;
+
+    /** Replicas. */
+    private final ConcurrentHashMap<String, Replica> replicas = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor for replica service.
+     *
+     * @param clusterNetSvc                 Cluster network service.
+     * @param mvDataStorage                 Partition storage.
+     */
+    public ReplicaManager(ClusterService clusterNetSvc, MvPartitionStorage mvDataStorage) {
+        this.clusterNetSvc = clusterNetSvc;
+        this.mvDataStorage = mvDataStorage;
+
+        clusterNetSvc.messagingService().addMessageHandler(ReplicaMessageGroup.class, new NetworkMessageHandler() {
+            @Override
+            public void onReceived(NetworkMessage message, NetworkAddress senderAddr, @Nullable Long correlationId) {
+                if (!busyLock.enterBusy()) {
+                    throw new IgniteException(new NodeStoppingException());
+                }
+
+                try {
+                    assert message instanceof ReplicaRequest : IgniteStringFormatter.format("Unexpected message [message={}]", message);
+
+                    ReplicaRequest request = (ReplicaRequest) message;
+
+                    Replica replica = replicas.get(request.partitionId());
+
+                    if (replica == null) {
+                        //TODO:IGNITE-17255 Sent an exceptional response to the client side when the replica is absent.
+                    }
+
+                    ReplicaResponse resp = replica.processRequest(request);
+
+                    clusterNetSvc.messagingService().respond(senderAddr, resp, correlationId);
+                } finally {
+                    busyLock.leaveBusy();
+                }
+            }
+        });
+    }
+
+    /**
+     * Gets a replica by its partition id.
+     *
+     * @param partitionId Partition id.
+     * @return Instance of the replica or {@code null} if the replica is not started.
+     */
+    public Replica replica(String partitionId) {
+        if (!busyLock.enterBusy()) {
+            throw new IgniteException(new NodeStoppingException());
+        }
+
+        try {
+            return replicas.get(partitionId);
+        } finally {
+            busyLock.leaveBusy();
+        }
+    }
+
+    /**
+     * Starts replica server.
+     *
+     * @param partId     Partition id.
+     * @param raftClient Raft client.
+     * @return Replica.
+     */
+    public Replica startReplica(String partId, RaftGroupService raftClient) {
```

Comment:
> > but I don't want parametrized this method with a listener, because we have only one implementation.
Please read my comments about different storages above.
Besides that we will have multiple implementations - e.g. partition specific and meta storage specific.

>If that method will be required in the future, 
It's already clear that we need it right now and not in future.

And what about
>  Is that method thread-safe: what will happen if I'll try to create several listeners for given partId at the same time?

Reply:
Today here are only interfaces